packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -31,52 +52,63 @@ local BLOOD_CONFIGS = {
 	},
 
 	-- shotgun
-	[536870914] = {
+	{
+		ClassName = "weapon_shotgun",
 		MinDensity = 100,
 		MaxDensity = 200,
-		VelocityPower = 14,
-		Randomness = 0.1,
-		CubeSize = 5,
+		VelocityPower = 5,
+		Randomness = 0.5,
+		CubeSize = 15,
+		DamageMultiplier = 35,
 	},
-
-	[DMG_SLASH] = {
+	{
+		ClassName = "weapon_357",
 		MinDensity = 100,
 		MaxDensity = 200,
-		VelocityPower = 4,
+		VelocityPower = 5,
 		Randomness = 0.1,
-		CubeSize = 3,
+		CubeSize = 5,
+		DamageMultiplier = 30,
 	},
+}
 
-	[DMG_CLUB] = {
-		MinDensity = 100,
-		MaxDensity = 200,
-		VelocityPower = 4,
-		Randomness = 0.1,
-		CubeSize = 3,
-	},
+local function getConfig(damageType)
+	for _, config in ipairs(BLOOD_CONFIGS) do
+        if bit.band(damageType, config.DamageFlags) ~= 0 then
+            return config
+        end
+	end
+	return nil
+end
 
-	[DMG_BLAST] = {
-		MinDensity = 3200,
-		MaxDensity = 5200,
-		VelocityPower = 40,
-		Randomness = 1,
-		CubeSize = 10,
-		DamageMultiplier = 90,
-	},
-}
+local function getWeaponConfig(weaponClassName)
+	for _, config in ipairs(BLOOD_WEAPON_CONFIGS) do
+		if weaponClassName == config.ClassName then
+			return config
+		end
+	end
+	return nil
+end
 
 local function sprayBlood(damageType, attacker, position, force, damage)
-	local normal = force:GetNormalized()
+	local normal = Vector()
 	if attacker == LocalPlayer() then
 		normal = LocalPlayer():GetAimVector()
+	else
+		normal = force:GetNormalized()
 	end
 
-	local config = BLOOD_CONFIGS[damageType]
+	local config = nil
+	if attacker:IsNPC() or attacker:IsPlayer() then
+		config = getWeaponConfig(attacker:GetActiveWeapon():GetClass())
+	end
 
 	if not config then
-		return
+		config = getConfig(damageType)
 	end
```

Comment:
Having the business logic of which config types take precedence in this function isn't great separation of concerns

It would be cleaner to make `getConfig` take both the damage type and the active weapon class, and implement the logic to prioritise weapon configs there